velox/functions/sparksql/tests/InTest.cpp

```diff
@@ -194,6 +194,57 @@ TEST_F(InTest, Bool) {
   EXPECT_EQ(in<bool>(false, {false}), true);
 }
 
+TEST_F(InTest, shortDecimal) {
+  EXPECT_EQ(in<int64_t>(1, {1, 2}, DECIMAL(2, 1)), true);
+  EXPECT_EQ(in<int64_t>(2, {1, 2}, DECIMAL(10, 5)), true);
+  EXPECT_EQ(in<int64_t>(3, {1, 2}, DECIMAL(17, 11)), false);
+  EXPECT_EQ(in<int64_t>(std::nullopt, {1, 2}, DECIMAL(3, 2)), std::nullopt);
```

Comment:
Perhaps the null input cases could be removed as they test the common path of implementation which has been ensured.